packages/edit-post/src/components/text-editor/index.js

```diff
@@ -3,23 +3,26 @@
  */
 import { PostTextEditor, PostTitle } from '@wordpress/editor';
 import { IconButton } from '@wordpress/components';
-import { withDispatch } from '@wordpress/data';
+import { withDispatch, withSelect } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
 import { displayShortcut } from '@wordpress/keycodes';
+import { compose } from '@wordpress/compose';
 
-function TextEditor( { onExit } ) {
+function TextEditor( { onExit, isRichEditingEnabled } ) {
 	return (
 		<div className="edit-post-text-editor">
-			<div className="edit-post-text-editor__toolbar">
-				<h2>{ __( 'Editing Code' ) }</h2>
-				<IconButton
-					onClick={ onExit }
-					icon="no-alt"
-					shortcut={ displayShortcut.secondary( 'm' ) }
-				>
-					{ __( 'Exit Code Editor' ) }
-				</IconButton>
-			</div>
+			{ isRichEditingEnabled && (
+				<div className="edit-post-text-editor__toolbar">
+					<h2>{ __( 'Editing Code' ) }</h2>
```

Comment:
@youknowriad Should this headline maybe still be displayed (without the button)? See https://github.com/WordPress/gutenberg/issues/12182 for a related report.

Reply:
Sure no strong opinion.